table/server/master/src/main/java/alluxio/master/table/Partition.java

```diff
@@ -20,14 +20,72 @@
 import alluxio.table.common.transform.TransformPlan;
 
 import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.stream.Collectors;
 
 /**
  * The table partition class.
  */
 public class Partition {
   private final String mPartitionSpec;
   private final Layout mBaseLayout;
-  private Layout mTransformedLayout;
+  /** New transformation is pushed to the end of the queue. */
+  private final ConcurrentLinkedDeque<Transformation> mTransformations =
```

Comment:
o, for now, we can just keep a single transformation. I just thought to make the RPC proto repeated in-case we want to add to it later, without having to change the rpc signature.